torch/lib/THC/THCAtomics.cuh

```diff
@@ -64,41 +65,41 @@ struct AtomicAddIntegerImpl<T, 4> {
 template<typename T>
 struct AtomicAddIntegerImpl<T, 8> {
   inline __device__ void operator()(T *address, T val) {
-    unsigned long long * address_as_ui = (unsigned long long *) (address);
-    unsigned long long old = *address_as_ui;
-    unsigned long long newval;
-    unsigned long long assumed;
+    uint64_t * address_as_ui = (uint64_t *) (address);
+    uint64_t old = *address_as_ui;
+    uint64_t newval;
+    uint64_t assumed;
 
     do {
       assumed = old;
       newval = val +  (T)old;
-      old = atomicCAS(address_as_ui, assumed, newval);
+      old = atomicCAS((uint64_t *) address_as_ui, (uint64_t) assumed, (uint64_t) newval);
```

Comment:
nit:`address_as_ui` and `assumed` should already by `uint64_t` i.e. is the cast necessary?